### PR TITLE
docbook to html

### PR DESCRIPTION
**Pull request by yozhio**
Move dbk-files to one directory
Fix validation
Convert docbook to html

### CODE REVIEW

**Review by worldmind — COMMENTED** [review]
A bit too radical approach ) We don't need separate directories for essay, but having directories for book/chain fully make sense, and `dbk` dir is not needed, perhaps I was unclear, let me show how I would like to see it:
```
book.english/
├── 01_Map_and_Territory
│   ├── 01_Predictably_Wrong
│   │   ├── 01_Preface.dbk
│   │   ├── 02_Biases:_An_Introduction.dbk
│   │   ├── 03_Scope_Insensitivity.dbk
│   │   ├── 04_The_Martial_Art_of_Rationality.dbk
│   │   ├── 05_Availability.dbk
│   │   ├── 06_What's_a_Bias?.dbk
│   │   ├── 07_Burdensome_Details.dbk
│   │   ├── 08_What_Do_We_Mean_By_"Rationality"?.dbk
│   │   ├── 09_Planning_Fallacy.dbk
│   │   ├── 10_Why_Truth?.dbk
│   │   ├── 11_Feeling_Rational.dbk
│   │   └── 12_The_Lens_That_Sees_Its_Flaws.dbk
│   ├── 02_Fake_Beliefs
│   │   ├── 01_Making_Beliefs_Pay_Rent_(in_Anticipated_Experiences).dbk
│   │   ├── 02_A_Fable_of_Science_and_Politics.dbk
│   │   ├── 03_Belief_in_Belief.dbk
...
│   ├── 03_Noticing_Confusion
│   │   ├── 01_Focus_Your_Uncertainty.dbk
│   │   ├── 02_What_is_Evidence?.dbk
...
```
and we need to move `book.english` from `lesswrong.com` directory, because it's not a part of original site

**Review by worldmind — APPROVED** [review]
Look good, sent couple of comments to PM because I have too old browser here

File: config/html.xsl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version='1.0'?>
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
+
+<xsl:import href="/usr/share/xml/docbook/stylesheet/nwalsh/html/docbook.xsl"/>
```
Comment (worldmind) [inline config/html.xsl:4]: Let's use chunked html where every section is a separate file, example is [here](https://github.com/worldmind/ratiocracy.book/blob/main/xsl/html-params.xsl)

**Review by worldmind — APPROVED** [review]
Looks good, but let's change xsl config for get chunked html

File: scripts/lib/lw_links.json
Comment (worldmind) [inline scripts/lib/lw_links.json:1]: Do we need to keep it in repo?

File: scripts/lib/htmlbook.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            "modified_at",
+            "author",
+            "coauthors",
+            "tags",
+            "user",
+            "reviewer",
+        ]:
+            setattr(self, k, info.get(k))
+
+    def get_content(self: Self) -> str:
```
Comment (worldmind) [inline scripts/lib/htmlbook.py:41]: Good to comment why it's not in init method

**Review by worldmind — APPROVED** [review]
Not saying I got a good understanding of the flow, but looks good, added some comments

File: scripts/lib/lw_links.json
Comment (yozhio) [reply scripts/lib/lw_links.json:1]: This is a reference for comparing links of the old format and the current one. There is no easy universal way to match them all on the fly, so they are easier to store.

File: Makefile [hunk 3]
```diff
@@ -49,20 +52,50 @@ book:
...
+$(HTML_DIR)/$(CROSSLINKS_DB): $(addprefix $(HTML_DIR)/,$(CLDB_CHUNKS))
+	xsltproc --stringparam dbk_files "$(DBK_FILES)" \
+		--stringparam target_ext ".html" \
+		-o $@ $(CLDB_XSL) $(word 1, $(DBK_FILES)) >> $(LOG_FILE) 2>&1
+
+$(HTML_DIR)/%.db: $(BOOK_DIR)/%$(BOOK_EXT)
+	xsltproc -xinclude $(PROFILING_XSL) $< | \
+	xsltproc --stringparam targets.filename "$@" \
+		--stringparam collect.xref.targets "only" \
+		-xinclude $(FO_XSL) - >> $(LOG_FILE) 2>&1
```
[the remark this answers is not in the document]
Comment (yozhio) [reply Makefile:90]: no, this is typo

File: lesswrong.com/Rationality: A-Z/05_Mere_Goodness/02_Value_Theory/06_2-Place_and_1-Place_Words/content.html [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<p><a onclick="window.open(this.href, '_blank', 'width=300,height=406,scrollbars=no,resizable=no,toolbar=no,directories=no,location=no,menubar=no,status=no,left=0,top=0'); return false" href="../../../../book.english/img/2f0d638ee6de765ac7d0b7467447e08e.jpg"><img style="margin: 0px 0px 5px 15px; float: right; width: 100px; height: 134px;" title="Monsterwithgirl_2" src="../../../../book.english/img/2f0d638ee6de765ac7d0b7467447e08e.jpg" border="0" alt="Monsterwithgirl_2" /></a></p>
```
Comment (worldmind) [inline lesswrong.com/Rationality: A-Z/05_Mere_Goodness/02_Value_Theory/06_2-Place_and_1-Place_Words/content.html:1]: book.english or book.en?

**Review by worldmind — APPROVED** [review]
Looks good, we can merge it already, but perhaps we can add top-level index.html with links to all volumes.

File: lesswrong.com/Rationality: A-Z/05_Mere_Goodness/02_Value_Theory/06_2-Place_and_1-Place_Words/content.html (hunk 4 above)
Comment (yozhio) [reply lesswrong.com/Rationality: A-Z/05_Mere_Goodness/02_Value_Theory/06_2-Place_and_1-Place_Words/content.html:1]: I am trying to find "book.english" in the repository, but no luck. Try to regenerate docbooks (make valid_book), this looks like HTML-file generated from old dbk file.

Comment (yozhio) [reply lesswrong.com/Rationality: A-Z/05_Mere_Goodness/02_Value_Theory/06_2-Place_and_1-Place_Words/content.html:1]: New directory is book.en. It's removed part of content

**Pull request merged**